homeassistant/components/geniushub/climate.py

```diff
@@ -101,3 +146,21 @@ async def async_set_hvac_mode(self, hvac_mode: str) -> None:
     async def async_set_preset_mode(self, preset_mode: str) -> None:
         """Set a new preset mode."""
         await self._zone.set_mode(HA_PRESET_TO_GH.get(preset_mode, "timer"))
+
+    async def async_set_zone(self, **kwargs) -> None:
+        """Set a new mode for this zone."""
+        mode = kwargs.get(ATTR_ZONE_MODE)
+
+        # pylint: disable=protected-access
+        if mode == "footprint" and not self._zone._has_pir:
```

Comment:
It would be good to update the library so we don't need to access a private attribute.